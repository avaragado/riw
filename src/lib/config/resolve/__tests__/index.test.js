// @flow

import type { ConfigSparseWithSource } from '../../';
import configResolve from '../';

type Fixture = {
    name: string,
    in: ConfigSparseWithSource,
};

const fixtures: Fixture[] = [
    {
        name: '01 empty input config',
        in: {},
    },

    {
        name: '02 empty input config with configFile',
        in: {
            configFile: '/foo/bar/package.json',
        },
    },

    {
        name: '03 empty input config with rootDir',
        in: {
            rootDir: '/foo/bar',
        },
    },

    {
        name: '04 empty input config with both configFile and rootDir',
        in: {
            configFile: '/a/b/c/package.json',
            rootDir: '/foo/bar',
        },
    },
];

describe('lib/config/resolve', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            const received = configResolve(fixture.in);

            expect(received).toMatchSnapshot();
        });
    });
});
