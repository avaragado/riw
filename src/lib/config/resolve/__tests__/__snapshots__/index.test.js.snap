// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/config/resolve 01 empty input config 1`] = `
Object {
  "collateDir": "tmp/babel-plugin-react-intl",
  "defaultLocale": "en-US",
  "inputMode": "source",
  "outputMode": "file-per-locale",
  "rootDir": ".",
  "sourceDirs": Array [
    "src/**/*.js",
  ],
  "targetLocales": Array [],
  "todoFile": "src/locale/TODO-untranslated.json",
  "translationsDatabaseFile": "src/locale/riw-db.json",
  "translationsOutputFile": "src/locale/[locale].json",
}
`;

exports[`lib/config/resolve 02 empty input config with configFile 1`] = `
Object {
  "collateDir": "tmp/babel-plugin-react-intl",
  "configFile": "/foo/bar/package.json",
  "defaultLocale": "en-US",
  "inputMode": "source",
  "outputMode": "file-per-locale",
  "rootDir": "/foo/bar",
  "sourceDirs": Array [
    "src/**/*.js",
  ],
  "targetLocales": Array [],
  "todoFile": "src/locale/TODO-untranslated.json",
  "translationsDatabaseFile": "src/locale/riw-db.json",
  "translationsOutputFile": "src/locale/[locale].json",
}
`;

exports[`lib/config/resolve 03 empty input config with rootDir 1`] = `
Object {
  "collateDir": "tmp/babel-plugin-react-intl",
  "defaultLocale": "en-US",
  "inputMode": "source",
  "outputMode": "file-per-locale",
  "rootDir": "/foo/bar",
  "sourceDirs": Array [
    "src/**/*.js",
  ],
  "targetLocales": Array [],
  "todoFile": "src/locale/TODO-untranslated.json",
  "translationsDatabaseFile": "src/locale/riw-db.json",
  "translationsOutputFile": "src/locale/[locale].json",
}
`;

exports[`lib/config/resolve 04 empty input config with both configFile and rootDir 1`] = `
Object {
  "collateDir": "tmp/babel-plugin-react-intl",
  "configFile": "/a/b/c/package.json",
  "defaultLocale": "en-US",
  "inputMode": "source",
  "outputMode": "file-per-locale",
  "rootDir": "/foo/bar",
  "sourceDirs": Array [
    "src/**/*.js",
  ],
  "targetLocales": Array [],
  "todoFile": "src/locale/TODO-untranslated.json",
  "translationsDatabaseFile": "src/locale/riw-db.json",
  "translationsOutputFile": "src/locale/[locale].json",
}
`;
